### DISCUSSION TITLE
automating testing of installs on various Operating systems

### DISCUSSION THREAD

**Discussion by maxandersen** [jbangdev/jbang#1122] [Ideas]
I've kicked off an initial github action workflow that is intended to run every night to verify install and upgrades continues to work. We've had random issues with that over time so thought it was better to get it automated rather than discover it by accident.

For now its at https://github.com/jbangdev/jbang/actions/workflows/install-tests.yml and just tests OSX install and initital run.
Will try add windows and linux too.

**Comment by maxandersen** [comment]
One challenge to test is updates, cc @quintesse 
i..e 0.77 to latest.

I tried doing this: 

```
 curl https://github.com/jbangdev/jbang/releases/download/v0.77.0/jbang-0.77.0.zip --output jbang.zip
unzip jbang.zip
mv jbang-* jbang
./jbang version --update
```

and jbang will complain to update but using package manager. Is my only option to put it into ~/.jbang/bin to have it honor the update ?

**Reply by quintesse** [reply]
Yup, no in-place updating

**Reply by maxandersen** [reply]
so is there a way to easily install any jbang zip properly or do I have to simulate it ?

**Reply by quintesse** [reply]
No, there is no way (currently). Even if you'd do something like `jbang app install jbang` with the jbang from the zip it would download the latest and install that. We could perhaps do a small work-around where, if you add the `--offline` option, jbang would install a copy of itself (if it detects it's not running from the correct install location already).

**Reply by quintesse** [reply]
Created #1125 that implements that. So you can now do `jbang --offline app install jbang`.
